hypothesis-python/src/hypothesis/stateful.py

```diff
@@ -777,16 +792,27 @@ def print_start(self):
     def print_end(self):
         report(u"state.teardown()")
 
-    def print_step(self, step):
+    def print_step(self, step, result):
         rule, data = step
         data_repr = {}
         for k, v in data.items():
             data_repr[k] = self.__pretty(v)
         self.step_count = getattr(self, u"step_count", 0) + 1
+        # If the step has target bundles, and the result is a MultipleResults
+        # then we want to assign to multiple variables.
+        if isinstance(result, MultipleResults):
+            n_output_vars = len(result.values)
+        else:
+            n_output_vars = 1
+        output_assignment = (
+            u"%s = " % (", ".join(self.last_names(n_output_vars)),)
```

Comment:
Looking at this line, we're going to have a `SyntaxError` if `result` is `MultipleResults()` because we'll print an assignment with no left-hand side.

I think the solution is to make the condition `if self.targets and n_output_vars >= 1`.  This also gives us a very elegant way to handle the exception case - instead of defaulting to `result = None`, use `result = MultipleResults()` and we'll get the no-assignment print for free.

I think this really is the last round of review though! :sweat_smile:

Reply:
That's a very good point that I hadn't considered!

I implemented the change that you suggested, and I also added an extra test to ensure that this is the case.

> I think this really is the last round of review though!

Thank you for taking the time to review!